### ISSUE TITLE
[Windows] requestedDisplayResolution high causes Screen.screens return null and app convulsions when move window between screens with different display scaling

### ISSUE THREAD

**Issue by itlancer** [Gamua/Adobe-Runtime-Support#170]
## Problem Description
`<requestedDisplayResolution>high</requestedDisplayResolution>` causes `Screen.screens` return `null` and app "convulsions" when move window from one screen to another with different disaplay scaling.
More about display scaling (and how to change it): https://www.windowscentral.com/how-set-custom-display-scaling-setting-windows-10
It always happens with Windows devices that have multiple screens with different scaling factors.
For example one 2880x1800px screen has 200% scaling and another one 1920x1080px with 125% scaling.

Tested with multiple AIR versions even with latests AIR 32.0.0.89, AIR 32.0.0.144 beta and AIR 33.0.2.330 with multiple (hundreds) Windows 8.1 and Windows 10 devices with 32-bit and 64-bit AIR applications with captive runtime. With different screens and resolutions with different configurations from 800x600px to 4K monitors per each.
The same problem in all cases.
It works fine with macOS.

Changing `renderMode` in application manifest doesn't help.
Chaning `gpuPreference` in application manifest doesn't help.

## Steps to Reproduce
Launch code below with Windows device with two or more screens and move application window from one screen to another by mouse. Application trace `Screen.screens` array on `Event.RESIZE` event.
Application manifest must contains `<requestedDisplayResolution>high</requestedDisplayResolution>`.
Screens must have different scaling, for example 100% and 150%. Sometimes you need to do it a few times to "find problem area" (usually it near the bottom of the second screen) - it depends on screens configuration and its scaling factors.
Application example with sources attached.
[requested_display_resolution_high_screen_screens_bug.zip](https://github.com/Gamua/Adobe-Runtime-Support/files/3929336/requested_display_resolution_high_screen_screens_bug.zip)

```
package {
	import flash.display.Sprite;
	import flash.display.Screen;
	import flash.events.Event;
	
	public class RequestedDisplayResolutionHighScreenScreensBug extends Sprite {
		
		public function RequestedDisplayResolutionHighScreenScreensBug() {
			stage.nativeWindow.addEventListener(Event.RESIZE, resize);
		}
		
		private function resize(e:Event):void {
			trace(Screen.screens);//This line returns null sometimes when you move app window from one screen to another
		}
	}
}
```

**Actual Result:**
Application window "convulsions" with multiple (dozens) incorrect resizes, scaling etc when you move it near screens edge.
`Event.RESIZE` will be fired multiple (dozens) times.
`Screen.screens` sometimes returns `null` instead of array of two `Screen` objects.
Application window sometimes lose "chrome" style (become borderless). It not 100% reproducible with any Windows devices but happens in most of times with Windows 8.1.

**Expected Result:**
Application window correctly moved to another screen without convulsions.
`Event.RESIZE` will be fired one or few times or not be fired at all.
`Screen.screens` should always return array of two `Screen` objects in this case.
Application window should not lose "chrome" style.

## Known Workarounds
none

**Comment by radubirsan** [comment]
Looks like this will never be fixed

**Comment by JanFW** [comment]
We are also looking for a fix on this issue and hope Harmann may look into it.
Expecially on 4k monitors, which become more and more a standard, and a multi window desktop AIR app, users would strongly benefit from a fix!

**Comment by radubirsan** [comment]
We rolled out highDPI and we are receiving customer complaints. Should we rollback or should we wait for a fix to this issue. 
I would appreciate a response.

**Comment by ajwfrost** [comment]
We'd been trying to reproduce this recently. From our observations, when you drag a window from one monitor to another, there's a point at which the application takes on the scale factor of the new monitor - but the resizing of the application that this implies, then adjusts the top/left point of the window. All of this appears to be 'standard' behaviour (i.e. Windows-application specific rather than AIR-specific).

We've not managed to get into the 'convulsions' state as the updated window size is still then mostly within the monitor for which the size is appropriate. From the description I'm guessing that there's a spot where, after the window has resized itself to the right size for Screen 2, it then realises that most of its area is on Screen 1 and so it resizes itself back to the size needed for Screen 1 - but then it finds most of its area is now on Screen 2 and so it resizes itself again... ad infinitum?

If there are any reproduction hints e.g. Screen 1 at 100%, Screen 2 at 200% directly above this, drag the window from the left hand part of the status bar up the right hand edge of the screen, etc....? it would be useful!

thanks

**Comment by JanFW** [comment]
I believe that the contentsScaleFactor of the stage seems to be the issue here, since it can only have one value throughout the entire application. If both monitors are having a different windows monitor scaling and when the problem when moving the window occurs, the contentsScaleFactor changes to the one of the monitor where the window gets moved to.
If both windows trace its contents scale factor you can see that it changes exactly at that time where the windows gets resized.

Maybe this ticket helps for further understanding, since I think it is related: https://github.com/Gamua/Adobe-Runtime-Support/issues/309

**Comment by itlancer** [comment]
@ajwfrost, seems I figure out.

With latest AIR SDK 33.1.1.300 there is no more `null` returns by `Screen.screens` as described in the first message (as it was with earlier AIR versions).
Also I cannot reproduce application window convulsions when move it between screens with different display scaling, even with previous AIR versions. Maybe some latest Windows updates fix it and it was not AIR issue.

But with latest AIR SDK 33.1.1.300 there is a new issue. If application manifest contains `<requestedDisplayResolution>high</requestedDisplayResolution>` and you have 2 screens with **different display scaling** and one of that has **200%  or more** then moving window between screens cause that window "disappear" somewhere. You cannot use application window anymore or display it someway. Only application restart could "restore" window. Sometimes to reproduce it you should maximize window and try to move it to another screen. Nevermind which one is the main screen.
Tested even with "empty" AIR application: 
[requested_display_resolution_high_window_bug.zip](https://github.com/Gamua/Adobe-Runtime-Support/files/5502314/requested_display_resolution_high_window_bug.zip)
It works fine without `<requestedDisplayResolution>high</requestedDisplayResolution>`.
Also it works fine if both screens have the same display scaling.
Also it works fine if there is no screen with 200%+ display scaling.
There is no such issue with previous AIR versions.

How it looks like with 2 screens (main left 3072 * 1920 pixels screen with 200% and secondary right screen 1920 * 1080 pixels with 100% display scaling):
![2](https://user-images.githubusercontent.com/10899066/98394768-4c160380-206c-11eb-9c6d-ca3bc21ac1c0.gif)

**Comment by itlancer** [comment]
All these issues (`null` returns by `Screen.screens`, window convulsions and window "disappearing") fixed with latest AIR SDK 33.1.1.345.
Thank you!

Also it will be great to have a fix for related issue: https://github.com/Gamua/Adobe-Runtime-Support/issues/309
